File: F/1237-find-positive-integer-solution-for-a-given-equation/main.go
package main

import "fmt"

func findSolution(customFunction func(int, int) int, z int) [][]int {
	var ret [][]int
	for i := 1; i <= 1000; i++ {
		s, e := 1, 1000
		for s <= e {
			m := s + (e-s)/2
			v := customFunction(i, m)
			if v > z {
				e = m - 1
			} else if v < z {
				s = m + 1
			} else {
				ret = append(ret, []int{i, m})
				break
			}
		}
	}
	return ret
}

func main() {
	fmt.Println("a")
}

File: F/1237-find-positive-integer-solution-for-a-given-equation/main.rs
struct Solution {}

impl Solution {
  pub fn find_solution(customfunction: &CustomFunction, z: i32) -> Vec<Vec<i32>> {
    let mut ret: Vec<Vec<i32>> = Vec::new();
    (1..=1000).for_each(|x| {
      let mut start: i32 = 1;
      let mut end: i32 = 1000;
      while start <= end {
        let m = start + (end - start) / 2;
        let v = customfunction.f(x, m);
        if v < z {
          s = m + 1;
        } else if v > z {
          e = m - 1;
        } else {
          ret.push(vec![x, m]);
          break;
        }
      }
    });

    ret
  }
}
